src/test/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java

```
@@ -0,0 +1,566 @@
+        if (propertyDefaultValueTag == null) {
+            final String tokenText = constructSubTreeText(nodeLi,
+                    defaultValueNode.getIndex(), nodeLi.getChildren().length);
+            result = cleanDefaultTokensText(tokenText);
+        }
+        else {
+            result = getTextFromTag(propertyDefaultValueTag);
+        }
+        return result;
+    }
+
+    /**
+     * Get the violation message text for a specific key from the list item.
+     *
+     * @param nodeLi list item javadoc node
+     * @return violation message key text
+     */
+    private static String getViolationMessages(DetailNode nodeLi) {
+        return getTextFromTag(
+                getFirstChildOfType(nodeLi, JavadocTokenTypes.JAVADOC_INLINE_TAG, 0));
+    }
+
+    /**
+     * Get text from {@code JavadocTokenTypes.JAVADOC_INLINE_TAG}.
+     *
+     * @param nodeTag target javadoc tag
+     * @return text contained by the tag
+     */
+    private static String getTextFromTag(DetailNode nodeTag) {
+        String result = "";
+        if (nodeTag != null) {
+            result = QUOTE_PATTERN.matcher(
+                    getFirstChildOfType(nodeTag, JavadocTokenTypes.TEXT, 0)
+                            .getText().trim()).replaceAll("");
+        }
+        return result;
+    }
+
+    /**
+     * Returns the first child node which matches the provided {@code TokenType} and has the
+     * children index after the offset value.
+     *
+     * @param node parent node
+     * @param tokenType token type to match
+     * @param offset children array index offset
+     * @return the first child satisfying the conditions
+     */
+    private static DetailNode getFirstChildOfType(DetailNode node, int tokenType, int offset) {
+        return Arrays.stream(node.getChildren())
+                .filter(child -> child.getIndex() >= offset && child.getType() == tokenType)
+                .findFirst()
+                .orElse(null);
```

it is better to either throw an exception or return Optional for such case. I see no good handling for null values in invocations of `getFirstChildOfType` method